### PR TITLE
fix(api): workflow permission unique name

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-cds#20742.1 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20742.1 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#20742.2 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

File: engine/api/rbac/dao_rbac_workflow.go [hunk 1]
```diff
@@ -85,7 +86,17 @@ func HasRoleOnWorkflowAndUserID(ctx context.Context, db gorp.SqlExecutor, role s
 	if allWorkflowAllowed {
 		return true, nil
 	}
-	return sdk.IsInArray(workflowName, workflows), nil
+	for _, item := range workflows {
+		g := glob.New(item)
```
Comment (richardlt) [inline engine/api/rbac/dao_rbac_workflow.go:90]: Why not using regex here ?

**Review by yesnault — APPROVED** [review]

**Review by fsamin — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**